// @cloudbet/market-helper/marketUtils.js
import { MarketType } from './sports-core';
/** @description function to check if its relevant to a totals market */
export var isMarketTotals = function (marketKey) {
    return marketKey.includes('total');
};
/** @description function to check if its relevant to a handicap market */
export var isMarketHandicap = function (marketKey) {
    return marketKey.includes('handicap') || marketKey.includes('run_line');
};
/** @description function to check if its relevant to a correct score market */
export var isMarketCorrectScore = function (marketKey) {
    return marketKey.includes('correct_score');
};
var sportMarkets = {};
/** @description get all the markets of sport */
export var getAllMarketsBySport = function (sportKey) {
    if (sportMarkets[sportKey]) {
        return sportMarkets[sportKey] || [];
    }
    var markets = Object.values(MarketType).filter(function (value) {
        return value.split('.')[0] === sportKey;
    });
    sportMarkets[sportKey] = markets;
    return markets;
};
/** @description get all the markets of sport besides the outright market */
export var getAllEventMarketsBySport = function (sportKey) {
    return getAllMarketsBySport(sportKey).filter(function (value) {
        return !value.includes('outright');
    });
};
/** @description get all the outright market of sports */
export var getAllOutrightMarketsBySport = function (sportKey) {
    return getAllMarketsBySport(sportKey).filter(function (value) {
        return value.includes('outright');
    });
};
