// @cloudbet/market-helper/sports-core/marketEnums.js
var MarketType;
(function (MarketType) {
    MarketType["american_football_1x2_period_first_half"] = "american_football.1x2_period_first_half";
    MarketType["american_football_double_chance"] = "american_football.double_chance";
    MarketType["american_football_draw_no_bet"] = "american_football.draw_no_bet";
    MarketType["american_football_drive_play_play_type"] = "american_football.drive_play_play_type";
    MarketType["american_football_drive_play_total_yards_gained"] = "american_football.drive_play_total_yards_gained";
    MarketType["american_football_drive_result"] = "american_football.drive_result";
    MarketType["american_football_freetext"] = "american_football.freetext";
    MarketType["american_football_halftime_fulltime_result"] = "american_football.halftime_fulltime_result";
    MarketType["american_football_handicap"] = "american_football.handicap";
    MarketType["american_football_handicap_period_first_half"] = "american_football.handicap_period_first_half";
    MarketType["american_football_highest_scoring_quarter"] = "american_football.highest_scoring_quarter";
    MarketType["american_football_moneyline"] = "american_football.moneyline";
    MarketType["american_football_next_scoring_type"] = "american_football.next_scoring_type";
    MarketType["american_football_odd_even"] = "american_football.odd_even";
    MarketType["american_football_outright"] = "american_football.outright";
    MarketType["american_football_quarter_1x2"] = "american_football.quarter_1x2";
    MarketType["american_football_quarter_handicap"] = "american_football.quarter_handicap";
    MarketType["american_football_quarter_handicap_incl_ot"] = "american_football.quarter_handicap_incl_ot";
    MarketType["american_football_quarter_total"] = "american_football.quarter_total";
    MarketType["american_football_quarter_totals_incl_ot"] = "american_football.quarter_totals_incl_ot";
    MarketType["american_football_race_to_points"] = "american_football.race_to_points";
    MarketType["american_football_team_highest_scoring_quarter"] = "american_football.team_highest_scoring_quarter";
    MarketType["american_football_team_odd_even"] = "american_football.team_odd_even";
    MarketType["american_football_team_total_period_first_half"] = "american_football.team_total_period_first_half";
    MarketType["american_football_team_totals"] = "american_football.team_totals";
    MarketType["american_football_totals"] = "american_football.totals";
    MarketType["american_football_totals_period_first_half"] = "american_football.totals_period_first_half";
    MarketType["american_football_totals_period_second_half"] = "american_football.totals_period_second_half";
    MarketType["american_football_touchdown_scorer"] = "american_football.touchdown_scorer";
    MarketType["archery_outright"] = "archery.outright";
    MarketType["athletics_outright"] = "athletics.outright";
    MarketType["aussie_rules_1st_half_1x2"] = "aussie_rules.1st_half_1x2";
    MarketType["aussie_rules_1st_half_double_chance"] = "aussie_rules.1st_half_double_chance";
    MarketType["aussie_rules_1st_half_draw_no_bet"] = "aussie_rules.1st_half_draw_no_bet";
    MarketType["aussie_rules_1st_half_handicap"] = "aussie_rules.1st_half_handicap";
    MarketType["aussie_rules_1st_half_odd_even"] = "aussie_rules.1st_half_odd_even";
    MarketType["aussie_rules_1st_half_team_total"] = "aussie_rules.1st_half_team_total";
    MarketType["aussie_rules_1st_half_total"] = "aussie_rules.1st_half_total";
    MarketType["aussie_rules_double_chance"] = "aussie_rules.double_chance";
    MarketType["aussie_rules_draw_no_bet"] = "aussie_rules.draw_no_bet";
    MarketType["aussie_rules_handicap"] = "aussie_rules.handicap";
    MarketType["aussie_rules_match_odds"] = "aussie_rules.match_odds";
    MarketType["aussie_rules_odd_even"] = "aussie_rules.odd_even";
    MarketType["aussie_rules_outright"] = "aussie_rules.outright";
    MarketType["aussie_rules_player_disposals"] = "aussie_rules.player_disposals";
    MarketType["aussie_rules_player_goals"] = "aussie_rules.player_goals";
    MarketType["aussie_rules_player_handballs"] = "aussie_rules.player_handballs";
    MarketType["aussie_rules_player_kicks"] = "aussie_rules.player_kicks";
    MarketType["aussie_rules_player_marks"] = "aussie_rules.player_marks";
    MarketType["aussie_rules_player_tackles"] = "aussie_rules.player_tackles";
    MarketType["aussie_rules_quarter_1x2"] = "aussie_rules.quarter_1x2";
    MarketType["aussie_rules_quarter_draw_no_bet"] = "aussie_rules.quarter_draw_no_bet";
    MarketType["aussie_rules_quarter_handicap"] = "aussie_rules.quarter_handicap";
    MarketType["aussie_rules_quarter_team_total"] = "aussie_rules.quarter_team_total";
    MarketType["aussie_rules_quarter_total"] = "aussie_rules.quarter_total";
    MarketType["aussie_rules_quarter_winning_margin"] = "aussie_rules.quarter_winning_margin";
    MarketType["aussie_rules_team_total"] = "aussie_rules.team_total";
    MarketType["aussie_rules_totals"] = "aussie_rules.totals";
    MarketType["aussie_rules_winner"] = "aussie_rules.winner";
    MarketType["badminton_correct_score"] = "badminton.correct_score";
    MarketType["badminton_exact_games"] = "badminton.exact_games";
    MarketType["badminton_game_odd_even"] = "badminton.game_odd_even";
    MarketType["badminton_game_point_handicap"] = "badminton.game_point_handicap";
    MarketType["badminton_game_pointnr"] = "badminton.game_pointnr";
    MarketType["badminton_game_race_to_points"] = "badminton.game_race_to_points";
    MarketType["badminton_game_total_points"] = "badminton.game_total_points";
    MarketType["badminton_game_winner"] = "badminton.game_winner";
    MarketType["badminton_handicap"] = "badminton.handicap";
    MarketType["badminton_nr_games_by_extra_points"] = "badminton.nr_games_by_extra_points";
    MarketType["badminton_outright"] = "badminton.outright";
    MarketType["badminton_totals"] = "badminton.totals";
    MarketType["badminton_winner"] = "badminton.winner";
    MarketType["bandy_match_odds"] = "bandy.match_odds";
    MarketType["bandy_outright"] = "bandy.outright";
    MarketType["bandy_totals"] = "bandy.totals";
    MarketType["baseball_batter_runs"] = "baseball.batter_runs";
    MarketType["baseball_freetext"] = "baseball.freetext";
    MarketType["baseball_handicap_innings_1_to_5"] = "baseball.handicap_innings_1_to_5";
    MarketType["baseball_inning_match_decision"] = "baseball.inning_match_decision";
    MarketType["baseball_max_consecutive_runs"] = "baseball.max_consecutive_runs";
    MarketType["baseball_moneyline"] = "baseball.moneyline";
    MarketType["baseball_moneyline_innings_1_to_5"] = "baseball.moneyline_innings_1_to_5";
    MarketType["baseball_odd_even"] = "baseball.odd_even";
    MarketType["baseball_outright"] = "baseball.outright";
    MarketType["baseball_race_to_runs"] = "baseball.race_to_runs";
    MarketType["baseball_run_line"] = "baseball.run_line";
    MarketType["baseball_team_totals"] = "baseball.team_totals";
    MarketType["baseball_total_scoreless_innings"] = "baseball.total_scoreless_innings";
    MarketType["baseball_totals"] = "baseball.totals";
    MarketType["baseball_totals_innings_1_to_5"] = "baseball.totals_innings_1_to_5";
    MarketType["baseball_which_team_wins_the_rest_of_the_match"] = "baseball.which_team_wins_the_rest_of_the_match";
    MarketType["baseball_with_extra_inning"] = "baseball.with_extra_inning";
    MarketType["basketball_1st_half_1x2"] = "basketball.1st_half_1x2";
    MarketType["basketball_1st_half_draw_no_bet"] = "basketball.1st_half_draw_no_bet";
    MarketType["basketball_1st_half_team_total"] = "basketball.1st_half_team_total";
    MarketType["basketball_1st_half_winning_margin"] = "basketball.1st_half_winning_margin";
    MarketType["basketball_1x2"] = "basketball.1x2";
    MarketType["basketball_2nd_half_1x2"] = "basketball.2nd_half_1x2";
    MarketType["basketball_2nd_half_1x2_incl_ot"] = "basketball.2nd_half_1x2_incl_ot";
    MarketType["basketball_2nd_half_handicap"] = "basketball.2nd_half_handicap";
    MarketType["basketball_2nd_half_total"] = "basketball.2nd_half_total";
    MarketType["basketball_any_team_to_lead_by_points"] = "basketball.any_team_to_lead_by_points";
    MarketType["basketball_any_team_total_maximum_consecutive_points"] = "basketball.any_team_total_maximum_consecutive_points";
    MarketType["basketball_freetext"] = "basketball.freetext";
    MarketType["basketball_halftime_fulltime"] = "basketball.halftime_fulltime";
    MarketType["basketball_handicap"] = "basketball.handicap";
    MarketType["basketball_handicap_period_first_half"] = "basketball.handicap_period_first_half";
    MarketType["basketball_handicap_period_second_half"] = "basketball.handicap_period_second_half";
    MarketType["basketball_highest_scoring_quarter"] = "basketball.highest_scoring_quarter";
    MarketType["basketball_moneyline"] = "basketball.moneyline";
    MarketType["basketball_odd_even"] = "basketball.odd_even";
    MarketType["basketball_odd_even_period_first_half"] = "basketball.odd_even_period_first_half";
    MarketType["basketball_outright"] = "basketball.outright";
    MarketType["basketball_point_range"] = "basketball.point_range";
    MarketType["basketball_quarter_1x2"] = "basketball.quarter_1x2";
    MarketType["basketball_quarter_handicap"] = "basketball.quarter_handicap";
    MarketType["basketball_quarter_totals"] = "basketball.quarter_totals";
    MarketType["basketball_quarter_winning_margin"] = "basketball.quarter_winning_margin";
    MarketType["basketball_race_to_points"] = "basketball.race_to_points";
    MarketType["basketball_race_to_points_incl_ot"] = "basketball.race_to_points_incl_ot";
    MarketType["basketball_scoring_type_pointnr"] = "basketball.scoring_type_pointnr";
    MarketType["basketball_team_to_lead_by_points"] = "basketball.team_to_lead_by_points";
    MarketType["basketball_team_total_maximum_consecutive_points"] = "basketball.team_total_maximum_consecutive_points";
    MarketType["basketball_team_totals"] = "basketball.team_totals";
    MarketType["basketball_total_period_first_half"] = "basketball.total_period_first_half";
    MarketType["basketball_total_period_second_half"] = "basketball.total_period_second_half";
    MarketType["basketball_totals"] = "basketball.totals";
    MarketType["basketball_winning_margin"] = "basketball.winning_margin";
    MarketType["beach_soccer_match_odds"] = "beach_soccer.match_odds";
    MarketType["beach_soccer_outright"] = "beach_soccer.outright";
    MarketType["beach_soccer_totals"] = "beach_soccer.totals";
    MarketType["beach_volleyball_handicap"] = "beach_volleyball.handicap";
    MarketType["beach_volleyball_outright"] = "beach_volleyball.outright";
    MarketType["beach_volleyball_totals"] = "beach_volleyball.totals";
    MarketType["beach_volleyball_winner"] = "beach_volleyball.winner";
    MarketType["bowls_correct_score"] = "bowls.correct_score";
    MarketType["bowls_handicap"] = "bowls.handicap";
    MarketType["bowls_outright"] = "bowls.outright";
    MarketType["bowls_set_1x2"] = "bowls.set_1x2";
    MarketType["bowls_set_and_end_total"] = "bowls.set_and_end_total";
    MarketType["bowls_set_and_end_winner"] = "bowls.set_and_end_winner";
    MarketType["bowls_set_draw_no_bet"] = "bowls.set_draw_no_bet";
    MarketType["bowls_set_end_exact_points"] = "bowls.set_end_exact_points";
    MarketType["bowls_set_end_odd_even"] = "bowls.set_end_odd_even";
    MarketType["bowls_set_handicap"] = "bowls.set_handicap";
    MarketType["bowls_set_point"] = "bowls.set_point";
    MarketType["bowls_set_race_to_points"] = "bowls.set_race_to_points";
    MarketType["bowls_set_team_total"] = "bowls.set_team_total";
    MarketType["bowls_set_total"] = "bowls.set_total";
    MarketType["bowls_totals"] = "bowls.totals";
    MarketType["bowls_winner"] = "bowls.winner";
    MarketType["boxing_1x2"] = "boxing.1x2";
    MarketType["boxing_outright"] = "boxing.outright";
    MarketType["boxing_totals"] = "boxing.totals";
    MarketType["boxing_will_the_fight_go_the_distance"] = "boxing.will_the_fight_go_the_distance";
    MarketType["boxing_winner"] = "boxing.winner";
    MarketType["boxing_winner_exact_rounds"] = "boxing.winner_exact_rounds";
    MarketType["boxing_winner_round_range"] = "boxing.winner_round_range";
    MarketType["boxing_winning_method"] = "boxing.winning_method";
    MarketType["call_of_duty_correct_score"] = "call_of_duty.correct_score";
    MarketType["call_of_duty_handicap"] = "call_of_duty.handicap";
    MarketType["call_of_duty_map_winner"] = "call_of_duty.map_winner";
    MarketType["call_of_duty_outright"] = "call_of_duty.outright";
    MarketType["call_of_duty_totals"] = "call_of_duty.totals";
    MarketType["call_of_duty_winner"] = "call_of_duty.winner";
    MarketType["chess_handicap"] = "chess.handicap";
    MarketType["chess_match_odds"] = "chess.match_odds";
    MarketType["chess_outright"] = "chess.outright";
    MarketType["counter_strike_correct_score_in_maps"] = "counter_strike.correct_score_in_maps";
    MarketType["counter_strike_map_1st_half_winner"] = "counter_strike.map_1st_half_winner";
    MarketType["counter_strike_map_1x2"] = "counter_strike.map_1x2";
    MarketType["counter_strike_map_handicap"] = "counter_strike.map_handicap";
    MarketType["counter_strike_map_knife_round_winner"] = "counter_strike.map_knife_round_winner";
    MarketType["counter_strike_map_odd_even_rounds"] = "counter_strike.map_odd_even_rounds";
    MarketType["counter_strike_map_player_with_most_deaths"] = "counter_strike.map_player_with_most_deaths";
    MarketType["counter_strike_map_player_with_most_kills"] = "counter_strike.map_player_with_most_kills";
    MarketType["counter_strike_map_race_to_rounds"] = "counter_strike.map_race_to_rounds";
    MarketType["counter_strike_map_round_bomb_defused"] = "counter_strike.map_round_bomb_defused";
    MarketType["counter_strike_map_round_bomb_planted"] = "counter_strike.map_round_bomb_planted";
    MarketType["counter_strike_map_round_handicap"] = "counter_strike.map_round_handicap";
    MarketType["counter_strike_map_round_kill_handicap"] = "counter_strike.map_round_kill_handicap";
    MarketType["counter_strike_map_round_kill_number"] = "counter_strike.map_round_kill_number";
    MarketType["counter_strike_map_round_player_with_kills"] = "counter_strike.map_round_player_with_kills";
    MarketType["counter_strike_map_round_team_total_kills"] = "counter_strike.map_round_team_total_kills";
    MarketType["counter_strike_map_round_total_kills"] = "counter_strike.map_round_total_kills";
    MarketType["counter_strike_map_round_winner"] = "counter_strike.map_round_winner";
    MarketType["counter_strike_map_team_total"] = "counter_strike.map_team_total";
    MarketType["counter_strike_map_total_headshots"] = "counter_strike.map_total_headshots";
    MarketType["counter_strike_map_total_rounds"] = "counter_strike.map_total_rounds";
    MarketType["counter_strike_map_will_there_be_an_ace"] = "counter_strike.map_will_there_be_an_ace";
    MarketType["counter_strike_map_will_there_be_overtime"] = "counter_strike.map_will_there_be_overtime";
    MarketType["counter_strike_map_winner"] = "counter_strike.map_winner";
    MarketType["counter_strike_map_xth_kill"] = "counter_strike.map_xth_kill";
    MarketType["counter_strike_match_odds"] = "counter_strike.match_odds";
    MarketType["counter_strike_outright"] = "counter_strike.outright";
    MarketType["counter_strike_total_maps"] = "counter_strike.total_maps";
    MarketType["counter_strike_winner"] = "counter_strike.winner";
    MarketType["cricket_away_home_supremacy"] = "cricket.away_home_supremacy";
    MarketType["cricket_freetext"] = "cricket.freetext";
    MarketType["cricket_home_away_supremacy"] = "cricket.home_away_supremacy";
    MarketType["cricket_innings_total"] = "cricket.innings_total";
    MarketType["cricket_innings_total_v2"] = "cricket.innings_total_v2";
    MarketType["cricket_outright"] = "cricket.outright";
    MarketType["cricket_over_team_total"] = "cricket.over_team_total";
    MarketType["cricket_over_total_runs_spread"] = "cricket.over_total_runs_spread";
    MarketType["cricket_player_total"] = "cricket.player_total";
    MarketType["cricket_team_run_range"] = "cricket.team_run_range";
    MarketType["cricket_team_total_in_highest_scoring_over"] = "cricket.team_total_in_highest_scoring_over";
    MarketType["cricket_team_totals"] = "cricket.team_totals";
    MarketType["cricket_team_with_highest_score_at_dismmissal"] = "cricket.team_with_highest_score_at_dismmissal";
    MarketType["cricket_test_1x2"] = "cricket.test_1x2";
    MarketType["cricket_test_innings_team_total"] = "cricket.test_innings_team_total";
    MarketType["cricket_test_match_inning_over_total_runs_spread"] = "cricket.test_match_inning_over_total_runs_spread";
    MarketType["cricket_test_match_innings_over_team_total"] = "cricket.test_match_innings_over_team_total";
    MarketType["cricket_test_match_innings_team_totals"] = "cricket.test_match_innings_team_totals";
    MarketType["cricket_test_match_innings_totals"] = "cricket.test_match_innings_totals";
    MarketType["cricket_test_match_player_total"] = "cricket.test_match_player_total";
    MarketType["cricket_test_match_team_run_range"] = "cricket.test_match_team_run_range";
    MarketType["cricket_test_match_team_total_in_highest_scoring_over"] = "cricket.test_match_team_total_in_highest_scoring_over";
    MarketType["cricket_test_winner"] = "cricket.test_winner";
    MarketType["cricket_will_there_be_a_tie"] = "cricket.will_there_be_a_tie";
    MarketType["cricket_winner"] = "cricket.winner";
    MarketType["curling_outright"] = "curling.outright";
    MarketType["cycling_head2head"] = "cycling.head2head";
    MarketType["cycling_outright"] = "cycling.outright";
    MarketType["darts_180s_handicap"] = "darts.180s_handicap";
    MarketType["darts_correct_score_in_sets"] = "darts.correct_score_in_sets";
    MarketType["darts_handicap_legs"] = "darts.handicap_legs";
    MarketType["darts_handicap_sets"] = "darts.handicap_sets";
    MarketType["darts_most_180s"] = "darts.most_180s";
    MarketType["darts_odd_even_sets"] = "darts.odd_even_sets";
    MarketType["darts_outright"] = "darts.outright";
    MarketType["darts_set_correct_score_in_legs"] = "darts.set_correct_score_in_legs";
    MarketType["darts_set_leg_any_player_to_score_a_180"] = "darts.set_leg_any_player_to_score_a_180";
    MarketType["darts_set_leg_checkout_color"] = "darts.set_leg_checkout_color";
    MarketType["darts_set_leg_handicap"] = "darts.set_leg_handicap";
    MarketType["darts_set_leg_team_to_score_a_180"] = "darts.set_leg_team_to_score_a_180";
    MarketType["darts_set_leg_total_darts"] = "darts.set_leg_total_darts";
    MarketType["darts_set_leg_winner"] = "darts.set_leg_winner";
    MarketType["darts_set_most_180s"] = "darts.set_most_180s";
    MarketType["darts_set_odd_even_legs"] = "darts.set_odd_even_legs";
    MarketType["darts_set_team_total_180s"] = "darts.set_team_total_180s";
    MarketType["darts_set_total_180s"] = "darts.set_total_180s";
    MarketType["darts_set_total_legs"] = "darts.set_total_legs";
    MarketType["darts_set_which_player_wins_the_rest"] = "darts.set_which_player_wins_the_rest";
    MarketType["darts_set_winner"] = "darts.set_winner";
    MarketType["darts_team_total_180s"] = "darts.team_total_180s";
    MarketType["darts_total_180s"] = "darts.total_180s";
    MarketType["darts_total_legs"] = "darts.total_legs";
    MarketType["darts_total_sets"] = "darts.total_sets";
    MarketType["darts_winner"] = "darts.winner";
    MarketType["dota_2_correct_score_in_maps"] = "dota_2.correct_score_in_maps";
    MarketType["dota_2_map_10_minutes_total_kills"] = "dota_2.map_10_minutes_total_kills";
    MarketType["dota_2_map_aegis_handicap"] = "dota_2.map_aegis_handicap";
    MarketType["dota_2_map_handicap"] = "dota_2.map_handicap";
    MarketType["dota_2_map_kill_draw_no_bet"] = "dota_2.map_kill_draw_no_bet";
    MarketType["dota_2_map_kill_handicap"] = "dota_2.map_kill_handicap";
    MarketType["dota_2_map_player_with_highest_creep_score"] = "dota_2.map_player_with_highest_creep_score";
    MarketType["dota_2_map_race_to_level_xth"] = "dota_2.map_race_to_level_xth";
    MarketType["dota_2_map_race_to_xth_kills"] = "dota_2.map_race_to_xth_kills";
    MarketType["dota_2_map_race_to_xth_net_worth"] = "dota_2.map_race_to_xth_net_worth";
    MarketType["dota_2_map_total_aegis"] = "dota_2.map_total_aegis";
    MarketType["dota_2_map_total_kills"] = "dota_2.map_total_kills";
    MarketType["dota_2_map_total_tower"] = "dota_2.map_total_tower";
    MarketType["dota_2_map_tower_handicap"] = "dota_2.map_tower_handicap";
    MarketType["dota_2_map_winner"] = "dota_2.map_winner";
    MarketType["dota_2_map_xth_aegis"] = "dota_2.map_xth_aegis";
    MarketType["dota_2_map_xth_barracks"] = "dota_2.map_xth_barracks";
    MarketType["dota_2_map_xth_kill"] = "dota_2.map_xth_kill";
    MarketType["dota_2_map_xth_shrine"] = "dota_2.map_xth_shrine";
    MarketType["dota_2_map_xth_tower"] = "dota_2.map_xth_tower";
    MarketType["dota_2_match_odds"] = "dota_2.match_odds";
    MarketType["dota_2_outright"] = "dota_2.outright";
    MarketType["dota_2_total_maps"] = "dota_2.total_maps";
    MarketType["dota_2_will_there_be_a_rampage"] = "dota_2.will_there_be_a_rampage";
    MarketType["dota_2_will_there_be_an_ultra_kill"] = "dota_2.will_there_be_an_ultra_kill";
    MarketType["dota_2_winner"] = "dota_2.winner";
    MarketType["entertainment_outright"] = "entertainment.outright";
    MarketType["esport_arena_of_valor_outright"] = "esport_arena_of_valor.outright";
    MarketType["esport_arena_of_valor_winner"] = "esport_arena_of_valor.winner";
    MarketType["esport_bga_outright"] = "esport_bga.outright";
    MarketType["esport_fifa_asian_handicap"] = "esport_fifa.asian_handicap";
    MarketType["esport_fifa_asian_handicap_period_extratime"] = "esport_fifa.asian_handicap_period_extratime";
    MarketType["esport_fifa_asian_handicap_period_first_half"] = "esport_fifa.asian_handicap_period_first_half";
    MarketType["esport_fifa_asian_handicap_period_second_half"] = "esport_fifa.asian_handicap_period_second_half";
    MarketType["esport_fifa_booking_match_odds"] = "esport_fifa.booking_match_odds";
    MarketType["esport_fifa_booking_nr"] = "esport_fifa.booking_nr";
    MarketType["esport_fifa_both_teams_to_score"] = "esport_fifa.both_teams_to_score";
    MarketType["esport_fifa_corner_handicap"] = "esport_fifa.corner_handicap";
    MarketType["esport_fifa_corner_match_odds"] = "esport_fifa.corner_match_odds";
    MarketType["esport_fifa_corner_nr"] = "esport_fifa.corner_nr";
    MarketType["esport_fifa_correct_score"] = "esport_fifa.correct_score";
    MarketType["esport_fifa_correct_score_period_1st_half"] = "esport_fifa.correct_score_period_1st_half";
    MarketType["esport_fifa_correct_score_period_2nd_half"] = "esport_fifa.correct_score_period_2nd_half";
    MarketType["esport_fifa_double_chance"] = "esport_fifa.double_chance";
    MarketType["esport_fifa_draw_no_bet"] = "esport_fifa.draw_no_bet";
    MarketType["esport_fifa_exact_total_goals_period_first_half"] = "esport_fifa.exact_total_goals_period_first_half";
    MarketType["esport_fifa_exact_total_goals_period_second_half"] = "esport_fifa.exact_total_goals_period_second_half";
    MarketType["esport_fifa_goal_nr"] = "esport_fifa.goal_nr";
    MarketType["esport_fifa_halftime_fulltime_result"] = "esport_fifa.halftime_fulltime_result";
    MarketType["esport_fifa_match_odds"] = "esport_fifa.match_odds";
    MarketType["esport_fifa_match_odds_period_extratime"] = "esport_fifa.match_odds_period_extratime";
    MarketType["esport_fifa_match_odds_period_first_half"] = "esport_fifa.match_odds_period_first_half";
    MarketType["esport_fifa_match_odds_period_second_half"] = "esport_fifa.match_odds_period_second_half";
    MarketType["esport_fifa_outright"] = "esport_fifa.outright";
    MarketType["esport_fifa_penalty_shootout_odd_even"] = "esport_fifa.penalty_shootout_odd_even";
    MarketType["esport_fifa_team_clean_sheet"] = "esport_fifa.team_clean_sheet";
    MarketType["esport_fifa_team_total_goals"] = "esport_fifa.team_total_goals";
    MarketType["esport_fifa_team_total_goals_period_first_half"] = "esport_fifa.team_total_goals_period_first_half";
    MarketType["esport_fifa_team_total_goals_period_second_half"] = "esport_fifa.team_total_goals_period_second_half";
    MarketType["esport_fifa_team_win_to_nil"] = "esport_fifa.team_win_to_nil";
    MarketType["esport_fifa_total_booking_points"] = "esport_fifa.total_booking_points";
    MarketType["esport_fifa_total_bookings"] = "esport_fifa.total_bookings";
    MarketType["esport_fifa_total_corners"] = "esport_fifa.total_corners";
    MarketType["esport_fifa_total_goals"] = "esport_fifa.total_goals";
    MarketType["esport_fifa_total_goals_period_extratime"] = "esport_fifa.total_goals_period_extratime";
    MarketType["esport_fifa_total_goals_period_first_half"] = "esport_fifa.total_goals_period_first_half";
    MarketType["esport_fifa_total_goals_period_second_half"] = "esport_fifa.total_goals_period_second_half";
    MarketType["esport_free_fire_outright"] = "esport_free_fire.outright";
    MarketType["esport_king_of_glory_outright"] = "esport_king_of_glory.outright";
    MarketType["esport_king_of_glory_winner"] = "esport_king_of_glory.winner";
    MarketType["esport_nba2k_1st_half_1x2"] = "esport_nba2k.1st_half_1x2";
    MarketType["esport_nba2k_1st_half_draw_no_bet"] = "esport_nba2k.1st_half_draw_no_bet";
    MarketType["esport_nba2k_1st_half_team_total"] = "esport_nba2k.1st_half_team_total";
    MarketType["esport_nba2k_1st_half_winning_margin"] = "esport_nba2k.1st_half_winning_margin";
    MarketType["esport_nba2k_1x2"] = "esport_nba2k.1x2";
    MarketType["esport_nba2k_2nd_half_1x2"] = "esport_nba2k.2nd_half_1x2";
    MarketType["esport_nba2k_2nd_half_1x2_incl_ot"] = "esport_nba2k.2nd_half_1x2_incl_ot";
    MarketType["esport_nba2k_2nd_half_handicap"] = "esport_nba2k.2nd_half_handicap";
    MarketType["esport_nba2k_2nd_half_total"] = "esport_nba2k.2nd_half_total";
    MarketType["esport_nba2k_any_team_to_lead_by_points"] = "esport_nba2k.any_team_to_lead_by_points";
    MarketType["esport_nba2k_any_team_total_maximum_consecutive_points"] = "esport_nba2k.any_team_total_maximum_consecutive_points";
    MarketType["esport_nba2k_halftime_fulltime"] = "esport_nba2k.halftime_fulltime";
    MarketType["esport_nba2k_handicap"] = "esport_nba2k.handicap";
    MarketType["esport_nba2k_handicap_period_first_half"] = "esport_nba2k.handicap_period_first_half";
    MarketType["esport_nba2k_handicap_period_second_half"] = "esport_nba2k.handicap_period_second_half";
    MarketType["esport_nba2k_highest_scoring_quarter"] = "esport_nba2k.highest_scoring_quarter";
    MarketType["esport_nba2k_odd_even"] = "esport_nba2k.odd_even";
    MarketType["esport_nba2k_odd_even_period_first_half"] = "esport_nba2k.odd_even_period_first_half";
    MarketType["esport_nba2k_outright"] = "esport_nba2k.outright";
    MarketType["esport_nba2k_point_range"] = "esport_nba2k.point_range";
    MarketType["esport_nba2k_quarter_1x2"] = "esport_nba2k.quarter_1x2";
    MarketType["esport_nba2k_quarter_handicap"] = "esport_nba2k.quarter_handicap";
    MarketType["esport_nba2k_quarter_totals"] = "esport_nba2k.quarter_totals";
    MarketType["esport_nba2k_quarter_winning_margin"] = "esport_nba2k.quarter_winning_margin";
    MarketType["esport_nba2k_race_to_points"] = "esport_nba2k.race_to_points";
    MarketType["esport_nba2k_race_to_points_incl_ot"] = "esport_nba2k.race_to_points_incl_ot";
    MarketType["esport_nba2k_scoring_type_pointnr"] = "esport_nba2k.scoring_type_pointnr";
    MarketType["esport_nba2k_team_to_lead_by_points"] = "esport_nba2k.team_to_lead_by_points";
    MarketType["esport_nba2k_team_total_maximum_consecutive_points"] = "esport_nba2k.team_total_maximum_consecutive_points";
    MarketType["esport_nba2k_team_totals"] = "esport_nba2k.team_totals";
    MarketType["esport_nba2k_total_period_first_half"] = "esport_nba2k.total_period_first_half";
    MarketType["esport_nba2k_total_period_second_half"] = "esport_nba2k.total_period_second_half";
    MarketType["esport_nba2k_totals"] = "esport_nba2k.totals";
    MarketType["esport_nba2k_winner_incl_overtime"] = "esport_nba2k.winner_incl_overtime";
    MarketType["esport_nba2k_winning_margin"] = "esport_nba2k.winning_margin";
    MarketType["esport_valorant_handicap"] = "esport_valorant.handicap";
    MarketType["esport_valorant_totals"] = "esport_valorant.totals";
    MarketType["esport_valorant_winner"] = "esport_valorant.winner";
    MarketType["esport_warcraft_correct_score"] = "esport_warcraft.correct_score";
    MarketType["esport_warcraft_handicap"] = "esport_warcraft.handicap";
    MarketType["esport_warcraft_totals"] = "esport_warcraft.totals";
    MarketType["esport_warcraft_winner"] = "esport_warcraft.winner";
    MarketType["field_hockey_match_odds"] = "field_hockey.match_odds";
    MarketType["field_hockey_outright"] = "field_hockey.outright";
    MarketType["field_hockey_totals"] = "field_hockey.totals";
    MarketType["floorball_match_odds"] = "floorball.match_odds";
    MarketType["floorball_outright"] = "floorball.outright";
    MarketType["floorball_totals"] = "floorball.totals";
    MarketType["formula_1_outright"] = "formula_1.outright";
    MarketType["formula_e_outright"] = "formula_e.outright";
    MarketType["futsal_double_chance"] = "futsal.double_chance";
    MarketType["futsal_draw_no_bet"] = "futsal.draw_no_bet";
    MarketType["futsal_handicap"] = "futsal.handicap";
    MarketType["futsal_match_odds"] = "futsal.match_odds";
    MarketType["futsal_odd_even"] = "futsal.odd_even";
    MarketType["futsal_outright"] = "futsal.outright";
    MarketType["futsal_totals"] = "futsal.totals";
    MarketType["golf_outright"] = "golf.outright";
    MarketType["golf_winner"] = "golf.winner";
    MarketType["greyhound_exacta"] = "greyhound.exacta";
    MarketType["greyhound_outright"] = "greyhound.outright";
    MarketType["greyhound_place"] = "greyhound.place";
    MarketType["greyhound_quinella"] = "greyhound.quinella";
    MarketType["greyhound_show"] = "greyhound.show";
    MarketType["greyhound_trifecta"] = "greyhound.trifecta";
    MarketType["greyhound_trio"] = "greyhound.trio";
    MarketType["greyhound_win"] = "greyhound.win";
    MarketType["handball_1st_half_total"] = "handball.1st_half_total";
    MarketType["handball_double_chance"] = "handball.double_chance";
    MarketType["handball_draw_no_bet"] = "handball.draw_no_bet";
    MarketType["handball_halftime_fulltime_result"] = "handball.halftime_fulltime_result";
    MarketType["handball_handicap"] = "handball.handicap";
    MarketType["handball_match_odds"] = "handball.match_odds";
    MarketType["handball_odd_even"] = "handball.odd_even";
    MarketType["handball_outright"] = "handball.outright";
    MarketType["handball_team_total"] = "handball.team_total";
    MarketType["handball_totals"] = "handball.totals";
    MarketType["handball_winner"] = "handball.winner";
    MarketType["hearthstone_outright"] = "hearthstone.outright";
    MarketType["hearthstone_winner"] = "hearthstone.winner";
    MarketType["ice_hockey_1x2"] = "ice_hockey.1x2";
    MarketType["ice_hockey_1x2_both_teams_to_score"] = "ice_hockey.1x2_both_teams_to_score";
    MarketType["ice_hockey_all_periods_under"] = "ice_hockey.all_periods_under";
    MarketType["ice_hockey_both_teams_to_score"] = "ice_hockey.both_teams_to_score";
    MarketType["ice_hockey_correct_score"] = "ice_hockey.correct_score";
    MarketType["ice_hockey_correct_score_incl_overtime_and_penalties"] = "ice_hockey.correct_score_incl_overtime_and_penalties";
    MarketType["ice_hockey_double_chance"] = "ice_hockey.double_chance";
    MarketType["ice_hockey_exact_goals"] = "ice_hockey.exact_goals";
    MarketType["ice_hockey_goal_nr"] = "ice_hockey.goal_nr";
    MarketType["ice_hockey_handicap"] = "ice_hockey.handicap";
    MarketType["ice_hockey_handicap_regular_time"] = "ice_hockey.handicap_regular_time";
    MarketType["ice_hockey_highest_scoring_period"] = "ice_hockey.highest_scoring_period";
    MarketType["ice_hockey_last_goal"] = "ice_hockey.last_goal";
    MarketType["ice_hockey_outright"] = "ice_hockey.outright";
    MarketType["ice_hockey_period_1x2"] = "ice_hockey.period_1x2";
    MarketType["ice_hockey_period_both_teams_to_score"] = "ice_hockey.period_both_teams_to_score";
    MarketType["ice_hockey_period_correct_score"] = "ice_hockey.period_correct_score";
    MarketType["ice_hockey_period_handicap"] = "ice_hockey.period_handicap";
    MarketType["ice_hockey_period_last_team_to_score"] = "ice_hockey.period_last_team_to_score";
    MarketType["ice_hockey_period_total"] = "ice_hockey.period_total";
    MarketType["ice_hockey_period_which_team_to_score"] = "ice_hockey.period_which_team_to_score";
    MarketType["ice_hockey_team_clean_sheet"] = "ice_hockey.team_clean_sheet";
    MarketType["ice_hockey_team_exact_goals"] = "ice_hockey.team_exact_goals";
    MarketType["ice_hockey_team_to_score_in_all_periods"] = "ice_hockey.team_to_score_in_all_periods";
    MarketType["ice_hockey_team_total"] = "ice_hockey.team_total";
    MarketType["ice_hockey_total"] = "ice_hockey.total";
    MarketType["ice_hockey_totals"] = "ice_hockey.totals";
    MarketType["ice_hockey_which_team_to_score"] = "ice_hockey.which_team_to_score";
    MarketType["ice_hockey_which_team_wins_the_rest_of_the_match"] = "ice_hockey.which_team_wins_the_rest_of_the_match";
    MarketType["ice_hockey_winner"] = "ice_hockey.winner";
    MarketType["ice_hockey_winning_margin"] = "ice_hockey.winning_margin";
    MarketType["ice_hockey_winning_margin_incl_overtime_and_penalties"] = "ice_hockey.winning_margin_incl_overtime_and_penalties";
    MarketType["kabaddi_handicap"] = "kabaddi.handicap";
    MarketType["kabaddi_match_odds"] = "kabaddi.match_odds";
    MarketType["kabaddi_outright"] = "kabaddi.outright";
    MarketType["kabaddi_total"] = "kabaddi.total";
    MarketType["league_of_legends_correct_score_in_maps"] = "league_of_legends.correct_score_in_maps";
    MarketType["league_of_legends_map_handicap"] = "league_of_legends.map_handicap";
    MarketType["league_of_legends_map_kill_draw_no_bet"] = "league_of_legends.map_kill_draw_no_bet";
    MarketType["league_of_legends_map_kill_handicap"] = "league_of_legends.map_kill_handicap";
    MarketType["league_of_legends_map_winner"] = "league_of_legends.map_winner";
    MarketType["league_of_legends_map_xth_baron"] = "league_of_legends.map_xth_baron";
    MarketType["league_of_legends_map_xth_dragon"] = "league_of_legends.map_xth_dragon";
    MarketType["league_of_legends_map_xth_inhibitor"] = "league_of_legends.map_xth_inhibitor";
    MarketType["league_of_legends_map_xth_kill"] = "league_of_legends.map_xth_kill";
    MarketType["league_of_legends_map_xth_tower"] = "league_of_legends.map_xth_tower";
    MarketType["league_of_legends_match_odds"] = "league_of_legends.match_odds";
    MarketType["league_of_legends_outright"] = "league_of_legends.outright";
    MarketType["league_of_legends_total_maps"] = "league_of_legends.total_maps";
    MarketType["league_of_legends_winner"] = "league_of_legends.winner";
    MarketType["mma_1x2"] = "mma.1x2";
    MarketType["mma_freetext"] = "mma.freetext";
    MarketType["mma_outright"] = "mma.outright";
    MarketType["mma_totals"] = "mma.totals";
    MarketType["mma_will_the_fight_go_the_distance"] = "mma.will_the_fight_go_the_distance";
    MarketType["mma_winner"] = "mma.winner";
    MarketType["mma_winner_exact_rounds"] = "mma.winner_exact_rounds";
    MarketType["mma_winning_method"] = "mma.winning_method";
    MarketType["motorsport_freetext"] = "motorsport.freetext";
    MarketType["motorsport_head2head"] = "motorsport.head2head";
    MarketType["motorsport_outright"] = "motorsport.outright";
    MarketType["olympics_outright"] = "olympics.outright";
    MarketType["overwatch_outright"] = "overwatch.outright";
    MarketType["overwatch_winner"] = "overwatch.winner";
    MarketType["pesapallo_match_odds"] = "pesapallo.match_odds";
    MarketType["pesapallo_outright"] = "pesapallo.outright";
    MarketType["politics_outright"] = "politics.outright";
    MarketType["rainbow_six_map_winner"] = "rainbow_six.map_winner";
    MarketType["rainbow_six_outright"] = "rainbow_six.outright";
    MarketType["rainbow_six_winner"] = "rainbow_six.winner";
    MarketType["rocket_league_correct_score"] = "rocket_league.correct_score";
    MarketType["rocket_league_handicap"] = "rocket_league.handicap";
    MarketType["rocket_league_map_winner"] = "rocket_league.map_winner";
    MarketType["rocket_league_outright"] = "rocket_league.outright";
    MarketType["rocket_league_total_maps"] = "rocket_league.total_maps";
    MarketType["rocket_league_winner"] = "rocket_league.winner";
    MarketType["rugby_league_draw_no_bet"] = "rugby_league.draw_no_bet";
    MarketType["rugby_league_halftime_fulltime_result"] = "rugby_league.halftime_fulltime_result";
    MarketType["rugby_league_handicap"] = "rugby_league.handicap";
    MarketType["rugby_league_moneyline"] = "rugby_league.moneyline";
    MarketType["rugby_league_odd_even"] = "rugby_league.odd_even";
    MarketType["rugby_league_outright"] = "rugby_league.outright";
    MarketType["rugby_league_team_totals"] = "rugby_league.team_totals";
    MarketType["rugby_league_totals"] = "rugby_league.totals";
    MarketType["rugby_league_winner"] = "rugby_league.winner";
    MarketType["rugby_union_draw_no_bet"] = "rugby_union.draw_no_bet";
    MarketType["rugby_union_halftime_fulltime_result"] = "rugby_union.halftime_fulltime_result";
    MarketType["rugby_union_handicap"] = "rugby_union.handicap";
    MarketType["rugby_union_moneyline"] = "rugby_union.moneyline";
    MarketType["rugby_union_odd_even"] = "rugby_union.odd_even";
    MarketType["rugby_union_outright"] = "rugby_union.outright";
    MarketType["rugby_union_team_totals"] = "rugby_union.team_totals";
    MarketType["rugby_union_totals"] = "rugby_union.totals";
    MarketType["rugby_union_winner"] = "rugby_union.winner";
    MarketType["sailing_outright"] = "sailing.outright";
    MarketType["sailing_winner"] = "sailing.winner";
    MarketType["snooker_correct_score"] = "snooker.correct_score";
    MarketType["snooker_handicap"] = "snooker.handicap";
    MarketType["snooker_odd_even_frames"] = "snooker.odd_even_frames";
    MarketType["snooker_outright"] = "snooker.outright";
    MarketType["snooker_totals"] = "snooker.totals";
    MarketType["snooker_winner"] = "snooker.winner";
    MarketType["soccer_10_minutes_sending_off_from_to"] = "soccer.10_minutes_sending_off_from_to";
    MarketType["soccer_1st_half_corner_handicap"] = "soccer.1st_half_corner_handicap";
    MarketType["soccer_5_minutes_team_total_corners_from_to"] = "soccer.5_minutes_team_total_corners_from_to";
    MarketType["soccer_5_minutes_total_from_to"] = "soccer.5_minutes_total_from_to";
    MarketType["soccer_anytime_goalscorer"] = "soccer.anytime_goalscorer";
    MarketType["soccer_asian_handicap"] = "soccer.asian_handicap";
    MarketType["soccer_asian_handicap_period_extratime"] = "soccer.asian_handicap_period_extratime";
    MarketType["soccer_asian_handicap_period_first_half"] = "soccer.asian_handicap_period_first_half";
    MarketType["soccer_asian_handicap_period_second_half"] = "soccer.asian_handicap_period_second_half";
    MarketType["soccer_booking_match_odds"] = "soccer.booking_match_odds";
    MarketType["soccer_booking_nr"] = "soccer.booking_nr";
    MarketType["soccer_both_teams_to_score"] = "soccer.both_teams_to_score";
    MarketType["soccer_corner_handicap"] = "soccer.corner_handicap";
    MarketType["soccer_corner_match_odds"] = "soccer.corner_match_odds";
    MarketType["soccer_corner_nr"] = "soccer.corner_nr";
    MarketType["soccer_correct_score"] = "soccer.correct_score";
    MarketType["soccer_correct_score_period_1st_half"] = "soccer.correct_score_period_1st_half";
    MarketType["soccer_correct_score_period_2nd_half"] = "soccer.correct_score_period_2nd_half";
    MarketType["soccer_double_chance"] = "soccer.double_chance";
    MarketType["soccer_draw_no_bet"] = "soccer.draw_no_bet";
    MarketType["soccer_exact_total_goals_period_first_half"] = "soccer.exact_total_goals_period_first_half";
    MarketType["soccer_exact_total_goals_period_second_half"] = "soccer.exact_total_goals_period_second_half";
    MarketType["soccer_freetext"] = "soccer.freetext";
    MarketType["soccer_goal_nr"] = "soccer.goal_nr";
    MarketType["soccer_goal_strike_zone"] = "soccer.goal_strike_zone";
    MarketType["soccer_goalnr_goalscorer"] = "soccer.goalnr_goalscorer";
    MarketType["soccer_halftime_fulltime_result"] = "soccer.halftime_fulltime_result";
    MarketType["soccer_last_corner"] = "soccer.last_corner";
    MarketType["soccer_last_goalscorer"] = "soccer.last_goalscorer";
    MarketType["soccer_match_odds"] = "soccer.match_odds";
    MarketType["soccer_match_odds_period_extratime"] = "soccer.match_odds_period_extratime";
    MarketType["soccer_match_odds_period_first_half"] = "soccer.match_odds_period_first_half";
    MarketType["soccer_match_odds_period_second_half"] = "soccer.match_odds_period_second_half";
    MarketType["soccer_outright"] = "soccer.outright";
    MarketType["soccer_penalty_shootout_odd_even"] = "soccer.penalty_shootout_odd_even";
    MarketType["soccer_team_clean_sheet"] = "soccer.team_clean_sheet";
    MarketType["soccer_team_total_goals"] = "soccer.team_total_goals";
    MarketType["soccer_team_total_goals_period_first_half"] = "soccer.team_total_goals_period_first_half";
    MarketType["soccer_team_total_goals_period_second_half"] = "soccer.team_total_goals_period_second_half";
    MarketType["soccer_team_win_to_nil"] = "soccer.team_win_to_nil";
    MarketType["soccer_total_booking_points"] = "soccer.total_booking_points";
    MarketType["soccer_total_bookings"] = "soccer.total_bookings";
    MarketType["soccer_total_corners"] = "soccer.total_corners";
    MarketType["soccer_total_goals"] = "soccer.total_goals";
    MarketType["soccer_total_goals_period_extratime"] = "soccer.total_goals_period_extratime";
    MarketType["soccer_total_goals_period_first_half"] = "soccer.total_goals_period_first_half";
    MarketType["soccer_total_goals_period_second_half"] = "soccer.total_goals_period_second_half";
    MarketType["soccer_when_will_goal_be_scored_intervals"] = "soccer.when_will_goal_be_scored_intervals";
    MarketType["soccer_which_team_wins_the_rest_of_the_match"] = "soccer.which_team_wins_the_rest_of_the_match";
    MarketType["specials_outright"] = "specials.outright";
    MarketType["squash_correct_score"] = "squash.correct_score";
    MarketType["squash_outright"] = "squash.outright";
    MarketType["squash_point_handicap"] = "squash.point_handicap";
    MarketType["squash_total_points"] = "squash.total_points";
    MarketType["squash_winner"] = "squash.winner";
    MarketType["starcraft_correct_score_in_maps"] = "starcraft.correct_score_in_maps";
    MarketType["starcraft_map_handicap"] = "starcraft.map_handicap";
    MarketType["starcraft_map_winner"] = "starcraft.map_winner";
    MarketType["starcraft_outright"] = "starcraft.outright";
    MarketType["starcraft_total_maps"] = "starcraft.total_maps";
    MarketType["starcraft_winner"] = "starcraft.winner";
    MarketType["street_fighter_v_outright"] = "street_fighter_v.outright";
    MarketType["sumo_outright"] = "sumo.outright";
    MarketType["sumo_winner"] = "sumo.winner";
    MarketType["swimming_outright"] = "swimming.outright";
    MarketType["table_tennis_correct_score"] = "table_tennis.correct_score";
    MarketType["table_tennis_exact_games"] = "table_tennis.exact_games";
    MarketType["table_tennis_game_odd_even"] = "table_tennis.game_odd_even";
    MarketType["table_tennis_game_point"] = "table_tennis.game_point";
    MarketType["table_tennis_game_point_handicap"] = "table_tennis.game_point_handicap";
    MarketType["table_tennis_game_race_to_points"] = "table_tennis.game_race_to_points";
    MarketType["table_tennis_game_total_points"] = "table_tennis.game_total_points";
    MarketType["table_tennis_game_winner"] = "table_tennis.game_winner";
    MarketType["table_tennis_handicap"] = "table_tennis.handicap";
    MarketType["table_tennis_how_many_games_will_be_decided_by_extra_points"] = "table_tennis.how_many_games_will_be_decided_by_extra_points";
    MarketType["table_tennis_outright"] = "table_tennis.outright";
    MarketType["table_tennis_totals"] = "table_tennis.totals";
    MarketType["table_tennis_winner"] = "table_tennis.winner";
    MarketType["tennis_any_set_to_nil"] = "tennis.any_set_to_nil";
    MarketType["tennis_correct_score"] = "tennis.correct_score";
    MarketType["tennis_exact_points"] = "tennis.exact_points";
    MarketType["tennis_exact_sets"] = "tennis.exact_sets";
    MarketType["tennis_game_handicap"] = "tennis.game_handicap";
    MarketType["tennis_game_handicap_in_set"] = "tennis.game_handicap_in_set";
    MarketType["tennis_odd_even_games"] = "tennis.odd_even_games";
    MarketType["tennis_outright"] = "tennis.outright";
    MarketType["tennis_point_in_game_and_set"] = "tennis.point_in_game_and_set";
    MarketType["tennis_race_to_games_in_set"] = "tennis.race_to_games_in_set";
    MarketType["tennis_race_to_points_in_set_and_game"] = "tennis.race_to_points_in_set_and_game";
    MarketType["tennis_set_correct_score"] = "tennis.set_correct_score";
    MarketType["tennis_set_game_correct_score"] = "tennis.set_game_correct_score";
    MarketType["tennis_set_game_correct_score_or_break"] = "tennis.set_game_correct_score_or_break";
    MarketType["tennis_set_game_to_deuce"] = "tennis.set_game_to_deuce";
    MarketType["tennis_set_handicap"] = "tennis.set_handicap";
    MarketType["tennis_set_tiebreak_total_points"] = "tennis.set_tiebreak_total_points";
    MarketType["tennis_team_to_win_a_set"] = "tennis.team_to_win_a_set";
    MarketType["tennis_team_total_games"] = "tennis.team_total_games";
    MarketType["tennis_total_games"] = "tennis.total_games";
    MarketType["tennis_total_games_in_set"] = "tennis.total_games_in_set";
    MarketType["tennis_total_sets"] = "tennis.total_sets";
    MarketType["tennis_winner"] = "tennis.winner";
    MarketType["tennis_winner_in_game_and_set"] = "tennis.winner_in_game_and_set";
    MarketType["tennis_winner_of_set"] = "tennis.winner_of_set";
    MarketType["tennis_with_tiebreak"] = "tennis.with_tiebreak";
    MarketType["volleyball_correct_score"] = "volleyball.correct_score";
    MarketType["volleyball_handicap"] = "volleyball.handicap";
    MarketType["volleyball_outright"] = "volleyball.outright";
    MarketType["volleyball_set_handicap"] = "volleyball.set_handicap";
    MarketType["volleyball_total_sets"] = "volleyball.total_sets";
    MarketType["volleyball_totals"] = "volleyball.totals";
    MarketType["volleyball_winner"] = "volleyball.winner";
    MarketType["waterpolo_match_odds"] = "waterpolo.match_odds";
    MarketType["waterpolo_outright"] = "waterpolo.outright";
    MarketType["waterpolo_totals"] = "waterpolo.totals";
    MarketType["winter_sports_outright"] = "winter_sports.outright";
    MarketType["world_lottery_bonus_ball_equals_any_regular_ball"] = "world_lottery.bonus_ball_equals_any_regular_ball";
    MarketType["world_lottery_bonus_ball_odd_even"] = "world_lottery.bonus_ball_odd_even";
    MarketType["world_lottery_bonus_ball_range"] = "world_lottery.bonus_ball_range";
    MarketType["world_lottery_bonus_ball_single_digit"] = "world_lottery.bonus_ball_single_digit";
    MarketType["world_lottery_bonus_ball_sum_odd_even"] = "world_lottery.bonus_ball_sum_odd_even";
    MarketType["world_lottery_bonus_ball_sum_total"] = "world_lottery.bonus_ball_sum_total";
    MarketType["world_lottery_bonus_ball_total"] = "world_lottery.bonus_ball_total";
    MarketType["world_lottery_draw_sum_odd_even"] = "world_lottery.draw_sum_odd_even";
    MarketType["world_lottery_draw_sum_range"] = "world_lottery.draw_sum_range";
    MarketType["world_lottery_draw_sum_total"] = "world_lottery.draw_sum_total";
    MarketType["world_lottery_draw_sum_total_incl_bonus_ball"] = "world_lottery.draw_sum_total_incl_bonus_ball";
    MarketType["world_lottery_outright"] = "world_lottery.outright";
    MarketType["world_lottery_standard_bet"] = "world_lottery.standard_bet";
    MarketType["world_lottery_standard_bet_incl_bonus_ball_in_additional_drum"] = "world_lottery.standard_bet_incl_bonus_ball_in_additional_drum";
    MarketType["world_lottery_standard_bet_incl_bonus_ball_in_same_drum"] = "world_lottery.standard_bet_incl_bonus_ball_in_same_drum";
})(MarketType || (MarketType = {}));
export { MarketType };
